vaadin-quarkus-extension/pom.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.vaadin</groupId>
+        <artifactId>vaadin-platform-parent</artifactId>
+        <version>23.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>vaadin-quarkus-extension</artifactId>
+    <name>Vaadin Quarkus Extension</name>
+    <description>Dependency for a Quarkus project.</description>
+
+    <repositories>
+        <!-- The order of definitions matters. Explicitly defining central here to make sure it has the highest priority. -->
+
+        <!-- Main Maven repository -->
+        <repository>
+            <id>central</id>
+            <url>https://repo.maven.apache.org/maven2</url>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+        </repository>
+        <repository>
+            <id>vaadin-prereleases</id>
+            <url>https://maven.vaadin.com/vaadin-prereleases</url>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.vaadin</groupId>
+                <artifactId>vaadin-spring-bom</artifactId>
+                <type>pom</type>
+                <scope>import</scope>
+                <version>${project.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```

Comment:
The spring dependency shouldn't be needed

Reply:
this is not a dependency but dependency management, so this wont bring in the spring dependencies directly, unless you declare it in the dependency section. 

while.. this name is probably misleading.

Reply:
But quarkus doesn't need it. Quarkus CDI based so probably people will not want to have spring there.

Reply:
this one only offer the required dependencies, which have been listed under. 
for using this bom, we don't need to define separate versions in this pom. 

and this bom does not bring in any dependencies, so it doesn't change anything

Reply:
the dependency tree of this module is 
```--- maven-dependency-plugin:2.8:tree (default-cli) @ vaadin-quarkus-extension ---
[INFO] com.vaadin:vaadin-quarkus-extension:jar:23.0-SNAPSHOT
[INFO] +- com.vaadin:vaadin-quarkus:jar:1.0.0:compile
[INFO] |  +- io.quarkus:quarkus-arc:jar:2.3.0.Final:compile
[INFO] |  |  +- io.quarkus.arc:arc:jar:2.3.0.Final:compile
[INFO] |  |  |  +- jakarta.transaction:jakarta.transaction-api:jar:1.3.3:compile
[INFO] |  |  |  \- org.jboss.logging:jboss-logging:jar:3.4.2.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-core:jar:2.3.0.Final:compile
[INFO] |  |  |  +- jakarta.inject:jakarta.inject-api:jar:1.0:compile
[INFO] |  |  |  +- io.quarkus:quarkus-ide-launcher:jar:2.3.0.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-development-mode-spi:jar:2.3.0.Final:compile
[INFO] |  |  |  +- io.smallrye.config:smallrye-config:jar:2.5.1:compile
[INFO] |  |  |  |  \- io.smallrye.config:smallrye-config-core:jar:2.5.1:compile
[INFO] |  |  |  |     +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0:compile
[INFO] |  |  |  |     +- io.smallrye.common:smallrye-common-expression:jar:1.6.0:compile
[INFO] |  |  |  |     |  \- io.smallrye.common:smallrye-common-function:jar:1.6.0:compile
[INFO] |  |  |  |     +- io.smallrye.common:smallrye-common-constraint:jar:1.6.0:compile
[INFO] |  |  |  |     +- io.smallrye.common:smallrye-common-classloader:jar:1.6.0:compile
[INFO] |  |  |  |     +- org.ow2.asm:asm:jar:9.1:compile
[INFO] |  |  |  |     \- io.smallrye.config:smallrye-config-common:jar:2.5.1:compile
[INFO] |  |  |  +- org.jboss.logmanager:jboss-logmanager-embedded:jar:1.0.9:compile
[INFO] |  |  |  +- org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final:compile
[INFO] |  |  |  +- org.jboss.threads:jboss-threads:jar:3.4.2.Final:compile
[INFO] |  |  |  +- org.jboss.slf4j:slf4j-jboss-logmanager:jar:1.1.0.Final:compile
[INFO] |  |  |  +- org.graalvm.sdk:graal-sdk:jar:21.2.0:compile
[INFO] |  |  |  +- org.wildfly.common:wildfly-common:jar:1.5.4.Final-format-001:compile
[INFO] |  |  |  \- io.quarkus:quarkus-bootstrap-runner:jar:2.3.0.Final:compile
[INFO] |  |  |     \- io.smallrye.common:smallrye-common-io:jar:1.6.0:compile
[INFO] |  |  \- org.eclipse.microprofile.context-propagation:microprofile-context-propagation-api:jar:1.2:compile
[INFO] |  +- io.quarkus:quarkus-undertow:jar:2.3.0.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-vertx-http:jar:2.3.0.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-security-runtime-spi:jar:2.3.0.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-mutiny:jar:2.3.0.Final:compile
[INFO] |  |  |  |  +- io.quarkus:quarkus-smallrye-context-propagation:jar:2.3.0.Final:compile
[INFO] |  |  |  |  |  \- io.smallrye:smallrye-context-propagation:jar:1.2.0:compile
[INFO] |  |  |  |  |     +- io.smallrye:smallrye-context-propagation-api:jar:1.2.0:compile
[INFO] |  |  |  |  |     \- io.smallrye:smallrye-context-propagation-storage:jar:1.2.0:compile
[INFO] |  |  |  |  \- io.smallrye.reactive:mutiny-smallrye-context-propagation:jar:1.0.0:compile
[INFO] |  |  |  +- io.quarkus:quarkus-vertx-http-dev-console-runtime-spi:jar:2.3.0.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-vertx:jar:2.3.0.Final:compile
[INFO] |  |  |  |  +- io.quarkus:quarkus-netty:jar:2.3.0.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-codec-haproxy:jar:4.1.68.Final:compile
[INFO] |  |  |  |  +- io.quarkus:quarkus-jackson:jar:2.3.0.Final:compile
[INFO] |  |  |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.12.5:compile
[INFO] |  |  |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.12.5:compile
[INFO] |  |  |  |  |  \- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.12.5:compile
[INFO] |  |  |  |  +- io.smallrye.common:smallrye-common-annotation:jar:1.6.0:compile
[INFO] |  |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-core:jar:2.13.0:compile
[INFO] |  |  |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-runtime:jar:2.13.0:compile
[INFO] |  |  |  |  |  \- io.smallrye.reactive:vertx-mutiny-generator:jar:2.13.0:compile
[INFO] |  |  |  |  |     \- io.vertx:vertx-codegen:jar:4.1.3:compile
[INFO] |  |  |  |  \- io.smallrye:smallrye-fault-tolerance-vertx:jar:5.2.1:compile
[INFO] |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-web:jar:2.13.0:compile
[INFO] |  |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-web-common:jar:2.13.0:compile
[INFO] |  |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-auth-common:jar:2.13.0:compile
[INFO] |  |  |  |  \- io.smallrye.reactive:smallrye-mutiny-vertx-bridge-common:jar:2.13.0:compile
[INFO] |  |  |  \- io.vertx:vertx-web:jar:4.1.4:compile
[INFO] |  |  |     +- io.vertx:vertx-web-common:jar:4.1.4:compile
[INFO] |  |  |     +- io.vertx:vertx-auth-common:jar:4.1.4:compile
[INFO] |  |  |     \- io.vertx:vertx-bridge-common:jar:4.1.4:compile
[INFO] |  |  +- io.quarkus.security:quarkus-security:jar:1.1.4.Final:compile
[INFO] |  |  |  \- io.smallrye.reactive:mutiny:jar:0.4.3:compile
[INFO] |  |  |     \- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  |  +- jakarta.enterprise:jakarta.enterprise.cdi-api:jar:2.0.2:compile
[INFO] |  |  |  +- jakarta.el:jakarta.el-api:jar:3.0.3:compile
[INFO] |  |  |  \- jakarta.interceptor:jakarta.interceptor-api:jar:1.2.5:compile
[INFO] |  |  |     \- jakarta.ejb:jakarta.ejb-api:jar:3.2.6:compile
[INFO] |  |  +- io.quarkus.http:quarkus-http-servlet:jar:4.1.1:compile
[INFO] |  |  +- jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  +- io.quarkus.http:quarkus-http-vertx-backend:jar:4.1.1:compile
[INFO] |  |  |  +- io.vertx:vertx-core:jar:4.1.0:compile
[INFO] |  |  |  |  +- io.netty:netty-common:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-buffer:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-transport:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-handler:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-handler-proxy:jar:4.1.65.Final:compile
[INFO] |  |  |  |  |  \- io.netty:netty-codec-socks:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-codec-http2:jar:4.1.65.Final:compile
[INFO] |  |  |  |  +- io.netty:netty-resolver:jar:4.1.65.Final:compile
[INFO] |  |  |  |  \- io.netty:netty-resolver-dns:jar:4.1.65.Final:compile
[INFO] |  |  |  |     \- io.netty:netty-codec-dns:jar:4.1.65.Final:compile
[INFO] |  |  |  \- io.quarkus.http:quarkus-http-http-core:jar:4.1.1:compile
[INFO] |  |  \- io.quarkus.http:quarkus-http-core:jar:4.1.1:compile
[INFO] |  |     \- io.netty:netty-codec-http:jar:4.1.60.Final:compile
[INFO] |  |        \- io.netty:netty-codec:jar:4.1.60.Final:compile
[INFO] |  \- com.vaadin:flow-server:jar:10.0.0.alpha5:compile
[INFO] |     +- org.slf4j:slf4j-api:jar:1.7.34:compile
[INFO] |     +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] |     +- com.vaadin.external.gwt:gwt-elemental:jar:2.8.2.vaadin2:compile
[INFO] |     +- commons-fileupload:commons-fileupload:jar:1.3.3:compile
[INFO] |     +- commons-io:commons-io:jar:2.11.0:compile
[INFO] |     +- com.fasterxml.jackson.core:jackson-core:jar:2.13.1:compile
[INFO] |     +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.1:compile
[INFO] |     |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.1:compile
[INFO] |     +- org.jsoup:jsoup:jar:1.14.3:compile
[INFO] |     +- com.helger:ph-css:jar:6.4.4:compile
[INFO] |     |  \- com.helger.commons:ph-commons:jar:10.1.5:compile
[INFO] |     |     \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |     +- net.bytebuddy:byte-buddy:jar:1.12.6:compile
[INFO] |     +- com.vaadin.external:gentyref:jar:1.2.0.vaadin1:compile
[INFO] |     +- org.apache.commons:commons-compress:jar:1.21:compile
[INFO] |     \- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |        +- org.apache.httpcomponents:httpcore:jar:4.4.13:compile
[INFO] |        +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |        \- commons-codec:commons-codec:jar:1.11:compile
[INFO] +- com.vaadin:vaadin-core:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-dev-server:jar:10.0.0.alpha5:compile
[INFO] |  +- com.vaadin:flow-lit-template:jar:10.0.0.alpha5:compile
[INFO] |  +- com.vaadin:flow-polymer-template:jar:10.0.0.alpha5:compile
[INFO] |  +- com.vaadin:flow-push:jar:10.0.0.alpha5:compile
[INFO] |  |  \- com.vaadin.external.atmosphere:atmosphere-runtime:jar:2.7.3.slf4jvaadin3:compile
[INFO] |  +- com.vaadin:flow-client:jar:10.0.0.alpha5:compile
[INFO] |  +- com.vaadin:flow-html-components:jar:10.0.0.alpha5:compile
[INFO] |  +- com.vaadin:flow-data:jar:10.0.0.alpha5:compile
[INFO] |  |  \- javax.validation:validation-api:jar:2.0.1.Final:compile
[INFO] |  +- com.vaadin:flow-dnd:jar:10.0.0.alpha5:compile
[INFO] |  |  +- org.webjars.npm:vaadin__vaadin-mobile-drag-drop:jar:1.0.1:compile
[INFO] |  |  \- org.webjars.npm:mobile-drag-drop:jar:2.3.0-rc.2:compile
[INFO] |  +- com.vaadin:vaadin-lumo-theme:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-material-theme:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-accordion-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-avatar-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-button-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-checkbox-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-combo-box-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-custom-field-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-date-picker-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-date-time-picker-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-details-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-time-picker-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-select-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-dialog-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-form-layout-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-grid-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  |  \- org.apache.commons:commons-lang3:jar:3.0:compile
[INFO] |  +- com.vaadin:vaadin-icons-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-iron-list-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-virtual-list-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-list-box-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-login-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-messages-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-ordered-layout-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-progress-bar-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-radio-button-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-renderer-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-split-layout-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-tabs-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-text-field-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-upload-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-notification-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-app-layout-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  +- com.vaadin:vaadin-context-menu-flow:jar:23.0-SNAPSHOT:compile
[INFO] |  \- com.vaadin:vaadin-menu-bar-flow:jar:23.0-SNAPSHOT:compile
[INFO] \- com.vaadin:vaadin-core-jandex:jar:23.0-SNAPSHOT:compile
```

Reply:
or this is the way you want? 

https://github.com/vaadin/platform/pull/2699

Reply:
So you are saying that you have named the `vaadin-core-bom` `vaadin-spring-bom` that should really be renamed, but I would rather that this not name spring and just use `vaadin-bom`

Reply:
now i updated to use the `vaadin-bom` instead of `vaadin-spring-bom`, this includes more dependencies like collaboration-engine etc.